### ISSUE TITLE
ENH: code review

### ISSUE THREAD

**Issue by Xylambda** [Javicadserres/DNet#13]
## Importing errors
Some internal imports are not correctly handled:
1.  There is no `module` file in the current directory of `layers`; yet, an attemp to importing it is made at [this line](https://github.com/Javicadserres/DNet/blob/main/DNet/layers/flatten_layers.py#L2). You should write the entire path:
```python
from Dnet.module import Base
```
2. Same thing [here](https://github.com/Javicadserres/DNet/blob/main/DNet/layers/layer.py#L2) and [here](https://github.com/Javicadserres/DNet/blob/main/DNet/layers/linear_layers.py#L2)

3. I would import all classes into the `__init__` file of `layers` submodule:
```python
# relative subpackages import
from . import convolutional_layers
from . import flatten_layers
from . import layer
from . import linear_layers
from . import padding_layers
from . import pooling_layers

from .convolutional_layers import Conv2D
from .padding_layers import ConstantPad

# and so on...
```
That way, you import layers in a more natural way:
```python
from Dnet.layers import Conv2D
```

## Doc ENH
1. Some classes do not describe their attributes (all `self` variables that are not assigned using the constructor arguments). I would be helpful to know them. Example
```python
class Conv2D(ConvBase):
    """
    Applies a 2D convolution over an input signal composed of several
    input planes.
    """
    def __init__(
        self, 
        in_channels, 
        out_channels, 
        kernel_size, 
        stride, 
        padding,
        padding_constant=0
    ):
        """
        Parameters
        ----------
        in_channels : int

        .....
        Attributes
        -----------
        A_resize : numpy.array
                Description.
        Z : numpy.array
                Description.

        ..... And so on ....
        """
```

2. The `Base` class should be documented. The doc [is empty](https://github.com/Javicadserres/DNet/blob/main/DNet/module.py#L3). 

3. I do miss more work in the documentation of the `optimizers`. It is [almost empty](https://github.com/Javicadserres/DNet/blob/main/DNet/optimizers.py). It should have references to papers, articles, etc where to find the math for these algorithms. Use [torch.optim docs](https://pytorch.org/docs/stable/optim.html#torch.optim.Adadelta) as example. Same goes for [loss module](https://github.com/Javicadserres/DNet/blob/main/DNet/loss.py).

4. Some optimizers do not have the correct DOC. See [line](https://github.com/Javicadserres/DNet/blob/main/DNet/optimizers.py#L41) and [line](https://github.com/Javicadserres/DNet/blob/main/DNet/optimizers.py#L52). It lacks an extra return argument in the doc.

## Structure
1. The module [module.py](https://github.com/Javicadserres/DNet/blob/main/DNet/module.py) seems to belong to the `layers` submodule. The name is also a bit confusing. It should be called `base` or `layer_base`.
2. Loss [functions](https://github.com/Javicadserres/DNet/blob/main/DNet/loss.py) and [activartions](https://github.com/Javicadserres/DNet/blob/main/DNet/activations.py) can also be implemented as layers. The just don't need the `optimize` method. The code will look more compact and optimized that way. See PyTorch for more examples.
## Code
1. The empty methods of `Base` class should have a `raise NotImplementedError()`
2. The [name](https://github.com/Javicadserres/DNet/blob/main/DNet/model.py#L58) should be `loss`, which is the correct vocabulary. 

## Build
1. The [requirement file](https://github.com/Javicadserres/DNet/blob/main/requirements.txt) is not finished. It needs the requirements to replicate the environment.
2. `versioneer` is not properly set. It needs `setup.cfg` file. Check [instructions](https://github.com/python-versioneer/python-versioneer/blob/master/INSTALL.md).
3. The README is not complete. It needs description and instructions to download and install the library.

I overall think is a really awasome framework! Be sure to update the `import errors` There are many!

@Javicadserres

**Comment by Javicadserres** [comment]
Thanks a lot for the comments @Xylambda! I really appreciate the feedback.

I will work on the points you mention above and let you know when I have made all the correspoding changes!
